addons/voice/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleTTSService.java

```diff
@@ -0,0 +1,320 @@
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.eclipse.smarthome.config.core.ConfigurableService;
+import org.eclipse.smarthome.core.audio.AudioException;
+import org.eclipse.smarthome.core.audio.AudioFormat;
+import org.eclipse.smarthome.core.audio.AudioStream;
+import org.eclipse.smarthome.core.audio.FileAudioStream;
+import org.eclipse.smarthome.core.voice.TTSException;
+import org.eclipse.smarthome.core.voice.TTSService;
+import org.eclipse.smarthome.core.voice.Voice;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import static org.openhab.voice.googletts.internal.GoogleTTSService.*;
+
+/**
+ * Voice service implementation.
+ *
+ * @author Gabor Bicskei - Initial contribution
+ */
+@Component(configurationPid = SERVICE_PID, property = {
+        Constants.SERVICE_PID + "=" + SERVICE_PID,
+        ConfigurableService.SERVICE_PROPERTY_LABEL + "=" + SERVICE_NAME,
+        ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=" + SERVICE_CATEGORY + ":" + SERVICE_ID,
+        ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=" + SERVICE_CATEGORY})
+public class GoogleTTSService implements TTSService {
+    /**
+     * Service name
+     */
+    static final String SERVICE_NAME = "Google Cloud TTS Service";
+
+    /**
+     * Service id
+     */
+    static final String SERVICE_ID = "googletts";
+
+    /**
+     * Service category
+     */
+    static final String SERVICE_CATEGORY = "voice";
+
+    /**
+     * Service pid
+     */
+    static final String SERVICE_PID = "org.openhab." + SERVICE_CATEGORY + "." + SERVICE_ID;
+
+    /**
+     * Cache folder under $userdata
+     */
+    private static final String CACHE_FOLDER_NAME = "cache";
+
+    /**
+     * Configuration parameters
+     */
+    private static final String PARAM_SERVICE_ACCOUNT_KEY = "serviceAccountKey";
+    private static final String PARAM_PITCH = "pitch";
+    private static final String PARAM_SPEAKING_RATE = "speakingRate";
+    private static final String PARAM_VOLUME_GAIN_DB = "volumeGainDb";
+
+    /**
+     * Logger.
+     */
+    private final Logger logger = LoggerFactory.getLogger(GoogleTTSService.class);
+
+    /**
+     * Set of supported audio formats
+     */
+    private Set<AudioFormat> audioFormats = new HashSet<>();
+
+    /**
+     * Google Cloud TTS API implementation
+     */
+    private GoogleCloudAPI apiImpl;
+
+    /**
+     * All voices for all supported locales
+     */
+    private Set<Voice> allVoices = new HashSet<>();
+
+    /**
+     * Audio format.
+     */
+    private GoogleTTSConfig config = new GoogleTTSConfig();
+
+    /**
+     * DS activate, with access to ConfigAdmin
+     */
+    @SuppressWarnings("unused")
```

Comment:
Is this still necessary? Eclipse shows "At least one of the problems in category 'unused' is not analysed due to a compiler option being ignored" and is still happy when I remove it.

Reply:
IntelliJ shows a warning without it... Want me to remove?

Reply:
What warning does it show?

It could be that your IntelliJ compiler warning settings may need some tuning so they exactly match the compiler warning settings used in the openHAB Eclipse IDE.

Reply:
It says the method is never used...

Reply:
Does the warning disappear when you annotate the method with `@Activate` (org.osgi.service.component.annotations.Activate) ?

Otherwise maybe [invalidating caches](https://intellij-support.jetbrains.com/hc/en-us/community/posts/360000110290-IDE-incorrectly-showing-classes-and-methods-as-unused) helps to resolve it?

Reply:
Yes these OSGi annotations also work with Eclipse. They are not strictly necessary. I think it doesn't hurt to add these annotations since I see them also used in a lot of other classes.